ledger/ledger.py

```diff
@@ -73,34 +73,28 @@ def recoverTree(self):
                             .format(type(self.tree)))
 
 
-        # ATTENTION!
-        # This functionality is disabled until better consistency verification
-        # implemented - always using recovery from transaction log
-        # from ledger.stores.memory_hash_store import MemoryHashStore
-        # from ledger.util import ConsistencyVerificationFailed
-        # if not self.tree.hashStore \
-        #         or isinstance(self.tree.hashStore, MemoryHashStore) \
-        #         or self.tree.leafCount == 0:
-        #     logging.info("Recovering tree from transaction log")
-        #     self.recoverTreeFromTxnLog()
-        # else:
-        #     try:
-        #         logging.info("Recovering tree from hash store of size {}".
-        #                       format(self.tree.leafCount))
-        #         self.recoverTreeFromHashStore()
-        #     except ConsistencyVerificationFailed:
-        #         logging.error("Consistency verification of merkle tree "
-        #                       "from hash store failed, "
-        #                       "falling back to transaction log")
-        #         self.recoverTreeFromTxnLog()
-
-        logging.debug("Recovering tree from transaction log")
+        from ledger.stores.memory_hash_store import MemoryHashStore
+        from ledger.util import ConsistencyVerificationFailed
         start = time.perf_counter()
-        self.recoverTreeFromTxnLog()
+        if not self.tree.hashStore \
+                or isinstance(self.tree.hashStore, MemoryHashStore) \
```

Comment:
Well we store full subtree roots in CompactMerkleTree, i am not sure whether we will need to store them in case of other kinds of trees, but we don't have others as of now